File: core/src/engine/rts/model/Resource.scala
package engine.rts.model

import scala.collection.JavaConversions._
import java.util.ArrayList
import java.util.ArrayList
import java.util

object Resource{
    var resourceList=new ArrayList[Resource]()
    def getResourceNames():ArrayList[String]={
        var rList=new ArrayList[String];
        for(res<-resourceList){
            rList.add(res.getName())
        }
        return rList
    }
}


/**
 * This class represents all the resources that will be present in the game.
 * @param resourceMap
 */
class Resource(resourceMap:Map[Any,Any])  extends java.io.Serializable {

    val name=resourceMap.get("name").get.asInstanceOf[String] //Set the name of the resource
    val resourceFile=resourceMap.get("image").get.asInstanceOf[String]//Set the image for the resource
    val id=Resource.resourceList.size()
    Resource.resourceList.add(this)
    def getName():String=name
    def getResourceFile():String=resourceFile
    def getID():Integer=id
    def printData(){println(name+":"+resourceFile)}
}

File: core/src/engine/rts/model/Resource.java
package engine.rts.model;

import java.util.ArrayList;

import com.badlogic.gdx.utils.JsonValue;

//TODO:Make sure these fields are not public.
public class Resource implements java.io.Serializable{


    public static ArrayList<Resource> resourceList=new ArrayList<Resource>();
 
    
    public static ArrayList<String> getResourceNames(){
    	ArrayList<String> names=new ArrayList<String>();
    	for(Resource res: resourceList){
    		names.add(res.getName());
    	}
    	return names;
    }
 
    public final String name;
    public final String image;
    public final int id;
    
    public Resource(JsonValue v){
    	name=v.getString("name");
    	image=v.getString("image");
    	id=v.getInt("id");
    	resourceList.add(this);
    }
    
 
    public String getName(){
    	return name;
    }
    
    public String getResourceFile(){
    	return image;
    }
    
    public int getId(){
    	return id;
    }
    
    public void printData(){
    	System.out.println(name+":"+image);
    }
    
    
}